src/ucp/tag/eager_rcv.c

```diff
@@ -103,16 +108,35 @@ ucp_eager_tagged_handler(void *arg, void *data, size_t length, unsigned am_flags
                                                   recv_len, 0, 1);
             ucp_request_complete_tag_recv(req, status);
         } else {
-            eagerf_hdr                = data;
-            req->recv.tag.info.length =
-            req->recv.remaining       = eagerf_hdr->total_len;
+            uint64_t msg_id;
+
+            if ((hdr_len == sizeof(*hdr.eagerf)) ||
```

Comment:
IMO adding to switch to enable/disable compat mode adds a lot of new code and potential conflicts. could be easier to just modify existing headers to always work in "compat" mode (always have ep_ptr)